### PR TITLE
fix(disable): skip disabled check for partially disabled server groups

### PR DESCRIPTION
**Pull request by dreynaud**
Rolling red black and monitored deployments create disable server group stages with a desiredPercentage, which only disables some instances and not the whole server group. In that case, we need to skip the check that the server group is disabled.

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java [hunk 1]
```diff
@@ -64,6 +69,23 @@ public TaskResult execute(@NotNull StageExecution stage) {
...
+    if (desiredPercentage instanceof Integer) {
+      return ((Integer) desiredPercentage) < 100;
+    }
+
+    if (desiredPercentage instanceof String) {
+      return Integer.parseInt((String) desiredPercentage) < 100;
+    }
+
+    log.warn("Unexpected value in stage context, desiredPercentage={}", desiredPercentage);
+    return true;
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java:86]: `stage.mapTo()` (and a try/catch) would probably save these conversations. 

Should the default be a partial disable if there's something wonky in `desiredPercentage`? That doesn't seem right.

Comment (dreynaud) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java:86]: ah, I always assumed `stage.mapTo()` was Groovy shenanigans

The default sounds wonky but it really means "when in doubt, don't enforce anything"

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java [hunk 2]
```diff
@@ -64,6 +69,23 @@ public TaskResult execute(@NotNull StageExecution stage) {
     }
   }
 
+  private boolean isPartialDisable(Object desiredPercentage) {
```
Comment (jonsie) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java:72]: Could you add a comment here such that we know this behavior is for rolling redblack and monitored deployment strategies?

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java (hunk 1 above)
Comment (dreynaud) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java]: in f9a1dadbe717be6b35e9ac52d702d643bcfe6988

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java (hunk 2 above)
Comment (dreynaud) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java]: ya good point

Comment (dreynaud) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java]: in 13c608c5310f3dfc4c96cf11ad53907c4fbc4378

**Review by aravindmd — APPROVED** [review]

**Pull request merged**